packages/medusa/src/loaders/helpers/routing/index.ts

```diff
@@ -0,0 +1,566 @@
+      route: "",
+      priority: Infinity,
+    }
+
+    if (parentPath) {
+      childPath = childPath.replace(parentPath, "")
+    }
+
+    descriptor.relativePath = childPath
+
+    let routeToParse = childPath
+
+    const pathSegments = childPath.split("/")
+    const lastSegment = pathSegments[pathSegments.length - 1]
+
+    if (lastSegment.startsWith("route")) {
+      pathSegments.pop()
+      routeToParse = pathSegments.join("/")
+    }
+
+    descriptor.route = this.parseRoute(routeToParse)
+    descriptor.priority = calculatePriority(descriptor.route)
+
+    this.routesMap.set(childPath, descriptor)
+  }
+
+  protected async createMiddlewaresDescriptor({
+    dirPath,
+  }: {
+    dirPath: string
+  }) {
+    const files = await readdir(dirPath)
+
+    const middlewareFilePath = files
+      .filter((path) => {
+        if (
+          this.excludes.length &&
+          this.excludes.some((exclude) => exclude.test(path))
+        ) {
+          return false
+        }
+
+        return true
+      })
+      .find((file) => {
+        return file.replace(/\.[^/.]+$/, "") === MIDDLEWARES_NAME
+      })
+
+    if (!middlewareFilePath) {
+      log({
+        activityId: this.activityId,
+        message: `No middleware files found in ${dirPath}. Skipping middleware configuration.`,
+      })
+      return
+    }
+
+    const absolutePath = join(dirPath, middlewareFilePath)
+
+    try {
+      await import(absolutePath).then((imp) => {
+        const middlewaresConfig = imp.config as MiddlewaresConfig | undefined
+
+        if (!middlewaresConfig) {
+          log({
+            activityId: this.activityId,
+            message: `No middleware configuration found in ${absolutePath}. Skipping middleware configuration.`,
+          })
+          return
+        }
+
+        middlewaresConfig.routes = middlewaresConfig.routes?.map((route) => {
+          return {
+            ...route,
+            method: route.method ?? "USE",
+          }
+        })
+
+        const descriptor: GlobalMiddlewareDescriptor = {
+          config: middlewaresConfig,
+        }
+
+        this.validateMiddlewaresConfig(descriptor)
+
+        this.globalMiddlewaresDescriptor = descriptor
+      })
+    } catch (error) {
+      log({
+        activityId: this.activityId,
+        message: `Failed to load middleware configuration in ${absolutePath}. Skipping middleware configuration.`,
+      })
+
+      return
+    }
+  }
+
+  protected async createRoutesMap({
+    dirPath,
+    parentPath,
+  }: {
+    dirPath: string
+    parentPath?: string
+  }): Promise<void> {
+    await Promise.all(
+      await readdir(dirPath, { withFileTypes: true }).then((entries) => {
+        return entries
+          .filter((entry) => {
+            /**
+             * We exclude all files starting with an underscore
+             * as underscore is used to indicate that files are
+             * to be ignored.
+             */
+            if (entry.name.startsWith("_")) {
```

Comment:
suggestion: this not explained in the pr description. Should we add this kind of information for the doc? maybe we can add that to the excludes as well

Reply:
also, since we are specifically looking for `route` file. I think we can remove all those checks and only keep the one line 408. wdyt?